test/FunctionalTests/Microsoft.OData.Core.Tests/IntegrationTests/Evaluation/AutoComputePayloadMetadataInJsonIntegrationTests.cs

```diff
@@ -752,6 +752,210 @@ public void WritingDynamicComplexPropertyWithModelSpecifiedInFullMetadataMode_40
+                  "\"http://svc/Orders(2)\"," +
+                  "\"http://svc/Orders(3)\"," +
+                  "\"http://svc/Orders(4)\"" +
+                "]" +
+              "}", payload);
+        }
+
+        [Fact]
+        public void WritingNestedCollectionEntityReferenceLinksInResponse_ODataV4_Works()
+        {
+            // Arrange & Act
+            string payload = GetEntityReferenceLinksPayload(ODataVersion.V4, request: false, collection: true);
+
+            // Assert
+            Assert.Equal("{\"@odata.context\":\"http://svc/$metadata#Customers/$entity\"," +
+                "\"CustomerId\":7," +
+                "\"BillOrders\":[" +
+                    "{\"@odata.id\":\"Orders(2)\"}," +
+                    "{\"@odata.id\":\"Orders(3)\"}," +
+                    "{\"@odata.id\":\"Orders(4)\"}" +
+                  "]" +
+                "}", payload);
+        }
+
+        [Fact]
+        public void WritingNestedCollectionEntityReferenceLinksInRequest_ODataV401_Works()
+        {
+            // Arrange & Act
+            string payload = GetEntityReferenceLinksPayload(ODataVersion.V401, request: true, collection: true);
+
+            // Assert
+            Assert.Equal("{\"@context\":\"http://svc/$metadata#Customers/$entity\"," +
+                "\"CustomerId\":7," +
+                "\"BillOrders\":[" +
+                    "{\"@id\":\"Orders(2)\"}," +
+                    "{\"@id\":\"Orders(3)\"}," +
+                    "{\"@id\":\"Orders(4)\"}" +
+                  "]" +
+                "}", payload);
+        }
+
+        [Fact]
+        public void WritingNestedCollectionEntityReferenceLinksInResponse_ODataV401_Works()
+        {
+            // Arrange & Act
+            string payload = GetEntityReferenceLinksPayload(ODataVersion.V401, request: false, collection: true);
+
+            // Assert
+            Assert.Equal("{\"@context\":\"http://svc/$metadata#Customers/$entity\"," +
+                "\"CustomerId\":7," +
+                "\"BillOrders\":[" +
+                    "{\"@id\":\"Orders(2)\"}," +
+                    "{\"@id\":\"Orders(3)\"}," +
+                    "{\"@id\":\"Orders(4)\"}" +
+                  "]" +
+                "}", payload);
+        }
+
+        private string GetEntityReferenceLinksPayload(ODataVersion version, bool request, bool collection)
+        {
+            MemoryStream stream = new MemoryStream();
+            ODataWriter odataWriter = GetODataWriterForEntityReferenceLink(version, request, stream);
+
+            odataWriter.WriteStart(new ODataResource { Properties = new[] { new ODataProperty { Name = "CustomerId", Value = 7 } } });
+            if (collection)
+            {
+                odataWriter.WriteStart(new ODataNestedResourceInfo { Name = "BillOrders", IsCollection = true });
+                odataWriter.WriteEntityReferenceLink(new ODataEntityReferenceLink { Url = new Uri("http://svc/Orders(2)") });

```

Comment:
>Url [](start = 84, length = 3)

it's too bad the property is named "Url", and not "Id", but not something to change at this point. Maybe in 8.0.

Reply:
Can't agree more.